src/gpg/verify.py

```diff
@@ -0,0 +1,36 @@
+import sys
+import gpg
+import time
+
+
+def verify(signature_file, filename):
+    """
+    verify the signature
+    signature_file has the detached signature
+    filename has the name of the file which is signed
+    """
+    
+    c = gpg.Context()
+    
+    try:
+        _, result = c.verify(open(filename), open(signature_file))
+        verified = True
+    except gpg.errors.BadSignatures as e:
+        verified = False
+        print(e)
+
+    if verified is True:
+        for i in range(len(result.signatures)):
+            sign = result.signatures[i]
+            print(  
+                'Signature made {2}\n'
+                '\tusing key {1}\n'
+                'Good signature from "{0}"'
+                .format(c.get_key(sign.fpr).uids[0].uid,
+                sign.fpr, time.ctime(sign.timestamp))         
+            )
```

Comment:
A clear and simple output message to the user is important, in my opinion.

The Python way of sending output messages seems too ugly to me, compared to Bash.
Maybe there is some elegant way in Python too, I don't know.

Reply:
here are a couple of other nicer options:
```python
time=time.ctime(sign.timestamp))
fingerprint=sign.fpr
signee=c.get_key(sign.fpr).uids[0].uid

print('Signature made {time}'.format(time=time))
print('using key {fingerprint}'.format(fingerprint=fingerprint)
print('Good signature from "{signee}"'.format(signee=signee))
```

```python
time=time.ctime(sign.timestamp))
fingerprint=sign.fpr
signee=c.get_key(sign.fpr).uids[0].uid

print('Signature made' + time)
print('using key' + fingerprint)
print('Good signature from'+ signee)
```

If you want a simpler message we can have something like
```python
.
.
.
print("Good signature from {email} made using {fingerprint} at {time}")
.
.
.
```

Reply:
This is one place you need logging so you can make use of the formatter, something like: 
`formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')`

Reply:
> Which one will you prefer...

It is not a matter of what I prefer, but what is the best. Use your own judgement to make the output simple, clear and useful. And also try to keep the code clean and beautiful.

In the example above I think that you should print the whole identity of the signer (name + email) not only email.

Reply:
> Maybe there is some elegant way in Python too,

Check out this one: https://rosettacode.org/wiki/Here_document#Python
You can use triple quoted strings instead of heredoc.